compare-processing/scripts/differential_expression.R

```diff
@@ -0,0 +1,113 @@
+#
+# Command line usage:
+#   Rscript scripts/differential_expression.R \
+#      -e <GENE_EXPRESSION_TSV> \
+#      -a <ANNOTATION_DF_TSV> \
+#      -p <VENN_DIAGRAM_PNG> \
+#      -r <LIMMA_RESULTS_TABLE_TSV> \
+#      -t <LIMMA_TOPTABLE_TSV>
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+#### command line arguments ----------------------------------------------------
+
+library("optparse")
+
+option.list = list(
+  make_option(c("-e", "--expression"), type = "character", default = "",
+              help = "Input gene expression matrix (TSV)", 
+              metavar = "character"),
+  make_option(c("-a", "--annotation"), type = "character", default = "",
+              help = "Input annotation (TSV)", metavar = "character"),
+  make_option(c("-p", "--plot"), type = "character", default = "",
+              help = "Venn diagram output file name, 
+              including path information", 
+              metavar = "character"),
+  make_option(c("-r", "--results"), type = "character", default = "",
+              help = "Results table output file name, 
+              including path information", 
+              metavar = "character"),
+  make_option(c("-t", "--toptable"), type = "character", default = "",
+              help = "Top table output file name, 
+              including path information", 
+              metavar = "character")
+)
+
+opt.parser <- OptionParser(option_list = option.list)
+opt <- parse_args(opt.parser)
+
+exprs.file <- opt$expression
+annot.file <- opt$annotation
+venn.output <- opt$plot
+results.output <- opt$results
+toptable.output <- opt$toptable
+
+#### read in data --------------------------------------------------------------
+
+# gene expression matrix
+exprs.df <- readr::read_tsv(exprs.file)
+colnames(exprs.df)[1] <- "ENSEMBL"
+exprs.mat <- as.data.frame(exprs.df) %>%
+  tibble::column_to_rownames("ENSEMBL") %>%
+  as.matrix()
+
+# file generated during heatmap plotting
+annot.df <- readr::read_tsv(annot.file)
+# we'll want to remove the "-" character as it will cause problems downstream
+annot.df <- annot.df %>% 
+              dplyr::mutate(dnHIF1a = stringr::word(dnHIF1a, 1, sep = "-"), 
+                            amputated = stringr::word(amputated, 1, sep = "-"))
+
+#### differential expression analysis ------------------------------------------
+
+# following the 9.5 Interaction Models: 2 × 2 Factorial Designs section of the
+# limma user guide
+
+# vector that contains info about the amputation and transgenic status of
+# each sample, this is consistent with the limma documentation
+amp.tg <- paste(annot.df$dnHIF1a, annot.df$amputated, sep = ".")
+amp.tg <- factor(amp.tg, levels = c("wildtype.unamputated",
+                                    "wildtype.post",
+                                    "dnHIF1a.unamputated",
+                                    "dnHIF1a.post"))
+# design matrices and initial fit
+design <- model.matrix(~ 0 + amp.tg)
```

Comment:
The section of the limma user's guide says
> This approach is recommended for most users, because the contrasts that are being tested are
formed explicitly.

About this kind of approach vs. something more streamlined

Reply:
Okey dokey then.